Esri/arcgis-maps-sdk-swift-samples
# [New] Find route around barriers

## Description

This PR implements `Find route around barriers` in `Routing and Logistics` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-FindRouteAroundBarriers/Shared/Samples/Find%20route%20around%20barriers)

## Linked Issue(s)

- `swift/issues/4646`

## How To Test

- Select "Stops" and tap on the map to add stops to the route.
- Select "Barriers" and tap on the map tap on the map to add areas that can't be crossed by the route.
- Tap "Route" to find the route and display it.
- Tap the settings button to toggle preferences like find the best sequence or preserve the first or last stop.
- Tap the directions button to view a list of the directions and tap on a direction to see the route for the direction.
- Tap the trash button to remove all features associated with the current feature selection.

## Screenshots
![find-route-around-barriers-1](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/4797956c-b57a-4f2a-93e6-e24dd527e616) ![find-route-around-barriers-2](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/0ecba261-0b09-4952-bb35-719b4cbc95cd)
